myapp/app/controllers/task_controller.rb

```diff
@@ -0,0 +1,55 @@
+class TaskController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def new
+    @task = Task.new
+    @labels = Label.all
+  end
+
+  def create
+    @task = Task.new(task_params)
+    @task.user_id = 1
+    @task.user_type = 'Task'
+    if @task.save
+      flash[:success] = "Task was successfully created."
+      redirect_to task_index_path(@task)
+    else
+      render :new
+    end
+  end
+
+  def show
+    @task = Task.find(params[:id])
+    @labels = @task.labels
```

Comment:
You give `@labels` here, but `myapp/app/views/task/show.html.erb` does not show label.
You should choose to remove `@labels` here or show labels at `show.html.erb`

Reply:
Its better to remove according to this step